File: Assets/src/model/tasks/AttackGroundTask.cs
using System.Collections.Generic;
using UnityEngine;

[System.Serializable]
public class AttackGroundTask : TelegraphedTask {

  public AttackGroundTask(Actor actor, Vector2Int targetPosition, int turns = 0) : base(actor, turns, new AttackGroundBaseAction(actor, targetPosition)) {
    TargetPosition = targetPosition;
  }

  public Vector2Int TargetPosition { get; }
}

File: web/src/model/tasks/AttackGroundTask.ts
import { TelegraphedTask } from './TelegraphedTask';
import { AttackGroundBaseAction } from '../BaseAction';
import { Vector2Int } from '../../core/Vector2Int';
import type { Actor } from '../Actor';

export class AttackGroundTask extends TelegraphedTask {
  readonly targetPosition: Vector2Int;

  constructor(actor: Actor, targetPosition: Vector2Int, turns = 0) {
    super(actor, turns, new AttackGroundBaseAction(actor, targetPosition));
    this.targetPosition = targetPosition;
  }
}
